terraform/jupyterhub.tf

```diff
@@ -1,74 +1,12 @@
-resource "aws_iam_role" "ecs-service" {
-  assume_role_policy = <<EOF
-{
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Action": "sts:AssumeRole",
-            "Principal": {
-                "Service": "ecs.amazonaws.com"
-            },
-            "Effect": "Allow",
-            "Sid": ""
-        }
-    ]
-}
-EOF
-}
-
-resource "aws_iam_role_policy_attachment" "ecs-service" {
-  role       = "${aws_iam_role.ecs-service.name}"
-  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerServiceFullAccess"
-}
-
-resource "aws_iam_instance_profile" "ecs-service" {
-  role = "${aws_iam_role.ecs-service.name}"
-}
-
-resource "aws_launch_configuration" "jupyterhub" {
+resource "aws_spot_instance_request" "cheap_worker" {
```

Comment:
You can use `spot_price ` of `aws_launch_configuration` to have potentially "self-healing" pool of at least one instance (provided that the `spot_price` is high enough in the available AZs.

Reply:
I do not currently have a launch configuration for the EMR cluster.  Would creating a variable for the spot price have roughly the same effect that you had in mind?